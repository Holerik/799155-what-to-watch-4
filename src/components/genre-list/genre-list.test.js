// genre-list.test.js
import React from 'react';
import renderer from 'react-test-renderer';
import GenreList from './genre-list.jsx';
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import history from '../../history.js';
import {Router} from 'react-router-dom';

const mockStore = configureStore([]);

it(`GenreList should render genres list correctly`, () => {
  const genresList = [`All genres`, `Drama`, `Crime`, `Biography`, `Thriller`, `Action`, `Mystery`];
  const store = mockStore({
    DATA: {
      genresList,
    },
    ERROR: {
      message: ``,
      show: false,
    },
  });

  const tree = renderer
    .create(
        <Provider store={store}>
          <Router history={history}>
            <GenreList
              activeItem={0}
              currentActiveItem={0}
              listItems={genresList}
              onMouseOver={() => {}}
              onMouseClick={() => {}}
              maxItemsCount={genresList.length}
            />
          </Router>
        </Provider>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
