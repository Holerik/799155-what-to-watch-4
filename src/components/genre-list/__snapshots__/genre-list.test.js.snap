// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreList should render genres list correctly 1`] = `
<ul
  className="catalog__genres-list"
  onClick={[Function]}
  onMouseOver={[Function]}
>
  <li
    className="catalog__genres-item
                            catalog__genres-item--active"
  >
    <a
      className="catalog__genres-link"
      href="/"
      onClick={[Function]}
    >
      All genres
    </a>
  </li>
  <li
    className="catalog__genres-item
                            "
  >
    <a
      className="catalog__genres-link"
      href="/"
      onClick={[Function]}
    >
      Drama
    </a>
  </li>
  <li
    className="catalog__genres-item
                            "
  >
    <a
      className="catalog__genres-link"
      href="/"
      onClick={[Function]}
    >
      Crime
    </a>
  </li>
  <li
    className="catalog__genres-item
                            "
  >
    <a
      className="catalog__genres-link"
      href="/"
      onClick={[Function]}
    >
      Biography
    </a>
  </li>
  <li
    className="catalog__genres-item
                            "
  >
    <a
      className="catalog__genres-link"
      href="/"
      onClick={[Function]}
    >
      Thriller
    </a>
  </li>
  <li
    className="catalog__genres-item
                            "
  >
    <a
      className="catalog__genres-link"
      href="/"
      onClick={[Function]}
    >
      Action
    </a>
  </li>
  <li
    className="catalog__genres-item
                            "
  >
    <a
      className="catalog__genres-link"
      href="/"
      onClick={[Function]}
    >
      Mystery
    </a>
  </li>
</ul>
`;
